src/dlr_relayvm.cc

```diff
@@ -316,3 +325,57 @@ void RelayVMModel::GetOutputByName(const char* name, void* out) {
   int output_index = this->GetOutputIndex(name);
   this->GetOutput(output_index, out);
 }
+
+/*! \brief A helper function to transform string input using InputMappingCategoricalString. When
+ * this map is present in the metadata file, the user is expected to provide string inputs to
+ * SetDLRInput as 1-D vector. This function will interpret the user's input as JSON, apply the
+ * mapping to convert strings to numbers, and produce a numeric NDArray which is given to TVM for
+ * the model input.*/
+void RelayVMModel::SetStringInput(int index, const int64_t* shape, void* input, int dim) {
```

Comment:
Can we split this into multiples functions to improve readability

For example.
```

void SetStringInput(...){
  SetJsonInput(input); // sets input_json_
  InitInputArray(); // initilizes ndarray with correct shape and size
  ConverJsonInputToNDArray(); // copies json into ndarray
}
```